packages/svelte/src/reactivity/set.js

```diff
@@ -50,8 +27,14 @@ export class ReactiveSet extends Set {
 		if (DEV) new Set(value);
 
 		if (value) {
+			// Support set-like objects that have a keys() method
+			if (!Array.isArray(value)) {
+				// @ts-ignore
+				value = Array.from(value.keys());
+			}
```

Comment:
this isn't right. we only care about `keys` when calling methods like `union` — in the constructor, we iterate over the argument itself. this would fail if you did `new Set(map)` for example. fix inbound

Reply:
According the MDN, if you pass a map in, it does `.keys()` on the setlike object to populate the Set entries.